vochain/processarchive/p2p.go

```diff
@@ -0,0 +1,87 @@
+package processarchive
+
+import (
+	"context"
+	"encoding/base64"
+	"time"
+
+	ipfscrypto "github.com/libp2p/go-libp2p-core/crypto"
+	"go.vocdoni.io/dvote/data"
+	"go.vocdoni.io/dvote/log"
+)
+
+const (
+	publishInterval = 10 * time.Minute
+	ipnsKeyAlias    = "processarchivekey"
+)
+
+// AddKey adds a base64 encoded ECDSA 256bit private key or generates
+// a new one.
+func (p *ProcessArchive) AddKey(b64key string) error {
+	var ipnsPk []byte
+	var err error
+	if b64key == "" {
+		// if key already exist, just return
+		if _, err := p.GetKey(); err == nil {
+			return nil
+		}
+		// else generate a new key
+		ipnsPk = data.NewIPFSkey()
+	} else {
+		ipnsPk, err = base64.StdEncoding.DecodeString(b64key)
+		if err != nil {
+			return err
+		}
+	}
+	return p.ipfs.AddKeyToKeystore(ipnsKeyAlias, ipnsPk)
+}
+
+// GetKey fetch the base64 encoded IPFS private key used to
+// publish the IPNS record.
+func (p *ProcessArchive) GetKey() (string, error) {
+	pk, err := p.ipfs.Node.Repo.Keystore().Get(ipnsKeyAlias)
+	if err != nil {
+		return "", err
+	}
+	pkb, err := ipfscrypto.MarshalPrivateKey(pk)
+	if err != nil {
+		return "", err
+	}
+	return base64.StdEncoding.EncodeToString(pkb), nil
```

Comment:
I admit I'm not a fan of using base64 strings for keys everywhere. It would be a bit cleaner to just use the unmarshaled []byte directly internally, and only deal with marshaling and base64 when interacting with the user (reading/writing a config, reading a flag, writing a log...).

There are even nice and general APIs for this, which should be compatible with most flag/config libraries: https://golang.org/pkg/encoding/#TextMarshaler

You don't have to do this here if it feels like too much added work. I could take a look in the future as part of a "flag cleanup" refactor.

Reply:
> I could take a look in the future as part of a "flag cleanup" refactor.

if so, we should raise an issue about it, too

Reply:
Yes, lets raise an Issue.